test/algorithms/l0_smooth.jl

```diff
@@ -22,6 +22,18 @@
         # βmax > 1e4
         @test_throws ArgumentError L0Smooth(βmax=1e3)
 
+        # forwarddiff and forwarddiff!
+        mat_in = rand(3, 3, 3)
+        mat_out = similar(mat_in)
+        forwarddiff!(mat_out, mat_in, dims = 2)
+        @test mat_out == forwarddiff(mat_in, dims = 2)
+
+        # backdiff and backdiff!
+        mat_in = rand(3, 3, 3)
+        mat_out = similar(mat_in)
+        backdiff!(mat_out, mat_in, dims = 3)
+        @test mat_out == backdiff(mat_in, dims = 3)
+

```

Comment:
We can move these to `test/utils.jl`, and we also need to check the numerical result on e.g., `reshape(collect(1:9), 3, 3)`